### PR TITLE
Update `R-api.rst` to unblock the `build_doc_r` check

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

#4483 added `create_registered_model` and `delete_registered_model` for R, but didn't update `R-api.rst`. This PR updates `R-api.rst` to unblock the `build_doc_r` check.

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ jobs:
       - run:
           name: Show diff
           command: |
+            git add --intent-to-add .
             git --no-pager diff --exit-code
```
Comment (harupy) [inline .circleci/config.yml:59]: Added `git add --intent-to-add .` to capture newly-created files.

File: .circleci/dump-r-dependencies.R [hunk 2]
```diff
@@ -1,4 +0,0 @@
-print(R.version)
```
Comment (harupy) [inline .circleci/dump-r-dependencies.R:1]: Moved this file under `mlflow/R/mlflow`.

File: .circleci/config.yml [hunk 3]
```diff
@@ -53,17 +53,34 @@ jobs:
...
+
+            if [[ "$failed" == "1" ]]; then
+              cat $PATCH_FILE
+              MLFLOW_REPO_ID=136202695
+              PATCH_FILE_URL="https://${CIRCLE_BUILD_NUM}-${MLFLOW_REPO_ID}-gh.circle-artifacts.com/0/${PATCH_FILE}"
+              echo "========== Run the following commands to apply the diff above =========="
+              echo "PATCH_FILE_URL=\"$PATCH_FILE_URL\""
+              echo 'REDIRECTED_URL=$(curl -Ls -o /dev/null -w %{url_effective} $PATCH_FILE_URL)'
+              echo 'curl $REDIRECTED_URL | git apply --verbose'
+              echo "========================================================================"
```
Comment (harupy) [inline .circleci/config.yml:72]: This allow us to quickly apply the diff by simply copy & pasting the commands in the terminal.

Comment (harupy) [reply .circleci/config.yml:72]: The output looks like this:

```
% PATCH_FILE_URL="https://8927-136202695-gh.circle-artifacts.com/0/21ed7fa02120377bd091db2277396603c26d1180.patch"
$ REDIRECTED_URL=$(curl -Ls -o /dev/null -w %{url_effective} $PATCH_FILE_URL)
$ curl $REDIRECTED_URL | git apply --verbose
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2118  100  2118    0     0   2097      0  0:00:01  0:00:01 --:--:--  2097
Checking patch mlflow/R/mlflow/man/mlflow_create_registered_model.Rd...
Checking patch mlflow/R/mlflow/man/mlflow_delete_registered_model.Rd...
Applied patch mlflow/R/mlflow/man/mlflow_create_registered_model.Rd cleanly.
Applied patch mlflow/R/mlflow/man/mlflow_delete_registered_model.Rd cleanly.
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -53,17 +53,34 @@ jobs:
...
 
       - run:
-          name: Show diff
+          name: Check diff
           command: |
-            git --no-pager diff --exit-code
+            set +e
+            git add .
+            PATCH_FILE="<< pipeline.git.revision >>.patch"
+            git --no-pager diff --cached --exit-code > $PATCH_FILE
```
Comment (harupy) [inline .circleci/config.yml:61]: Add `git add .` and `--cached` to detect new files.

Comment (harupy) [reply .circleci/config.yml:61]: https://circleci.com/docs/2.0/pipeline-variables:

> `pipeline.git.revision`: the long (40-character) git SHA that is being built:

File: .circleci/config.yml [hunk 5]
```diff
@@ -53,17 +53,34 @@ jobs:
...
 
       - save_cache:
-          key: r-cache-{{ checksum ".circleci/R-version" }}-{{ checksum ".circleci/depends.Rds" }}
+          key: r-cache-{{ checksum "mlflow/R/mlflow/R-version" }}-{{ checksum "mlflow/R/mlflow/depends.Rds" }}
           paths:
             - /usr/local/lib/R/site-library
 
       - store_artifacts:
-          path: docs/source/R-api.rst
+          path: << pipeline.git.revision >>.patch
```
Comment (harupy) [inline .circleci/config.yml:83]: Uploaded the patch created in the check-diff step as an artifact.

File: .circleci/config.yml (hunk 3 above)
Comment (harupy) [reply .circleci/config.yml:72]: I think this approach is simpler than:

1. Open the Artifacts tab on CricleCI
2. Open the patch there and download it
3. Run `git apply /path/to/patch`

File: .circleci/config.yml [hunk 6]
```diff
@@ -38,13 +38,13 @@ jobs:
...
+          working_directory: mlflow/R/mlflow
           command: |
-            Rscript .circleci/dump-r-dependencies.R
-            ls .circleci
+            Rscript .dump-r-dependencies.R
 
       - restore_cache:
           keys:
-            - r-cache-{{ checksum ".circleci/R-version" }}
+            - r-cache-{{ checksum "mlflow/R/mlflow/R-version" }}
```
Comment (harupy) [inline .circleci/config.yml:47]: Now `R-version` and `depends.Rds` are created in `mlflow/R/mlflow`.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**